yang/frr-zebra.yang

```diff
@@ -0,0 +1,1887 @@
+      "Initial revision.";
+    reference
+      "ISO/IEC 10589:2002.";
+  }
+
+  typedef unix-timestamp {
+    type uint32;
+    units "seconds";
+    description
+      "An absolute time in seconds since the unix epoch.";
+  }
+
+  /*
+   * Multicast RPF mode configurable type
+   */
+  typedef mcast-rpf-lookup-mode {
+    type enumeration {
+      enum "none" {
+	value 0;
+	description
+	  "No mode set.";
+      }
+      enum "mrib-only" {
+	value 1;
+	description
+	  "Lookup in unicast RIB only.";
+      }
+      enum "urib-only" {
+	value 2;
+	description
+	  "Lookup in multicast RIB only.";
+      }
+      enum "mrib-then-urib" {
+	value 3;
+	description
+	  "Try multicast RIB first, fall back to unicast RIB.";
+      }
+      enum "lower-distance" {
+	value 4;
+	description
+	  "Lookup both unicast and mcast, use entry with lower distance.";
+      }
+      enum "longer-prefix" {
+	value 5;
+	description
+	  "Lookup both unicast and mcast, use entry with longer prefix.";
+      }
+    }
+    description
+      "Multicast RPF lookup behavior";
+  }
+
+  typedef nexthop-type {
+    type enumeration {
+      enum "ifindex" {
+	value 1;
+	description
+	  "Specific interface.";
+      }
+      enum "ip4" {
+	value 2;
+	description
+	  "IPv4 address.";
+      }
+      enum "ip4-ifindex" {
+	value 3;
+	description
+	  "IPv4 address and interface.";
+      }
+      enum "ip6" {
+	value 4;
+	description
+	  "IPv6 address.";
+      }
+      enum "ip6-ifindex" {
+	value 5;
+	description
+	  "IPv6 address and interface.";
+      }
+      enum "blackhole" {
+	value 6;
+	description
+	  "Unreachable or prohibited.";
+      }
+    }
+    description
+      "Nexthop types.";
+  }
+
+  typedef blackhole-type {
+    type enumeration {
+      enum "unspec" {
+	value 0;
+	description
+	  "Generic unreachable.";
+      }
+      enum "null" {
+	value 1;
+	description
+	  "Null type.";
+      }
+      enum "reject" {
+	value 2;
+	description
+	  "ICMP unreachable.";
+      }
+      enum "prohibited" {
+	value 3;
+	description
+	  "ICMP admin-prohibited.";
+      }
+    }
+    default "null";
+    description
+      "Nexthop blackhole types.";
+  }
+
+  typedef gateway-address {
+    type union {
+      type inet:ipv4-address;
+      type inet:ipv6-address;
+    }
+  }
+
+  /*
+   * Nexthop object
+   */
+  grouping frr-nexthop {
+    leaf nh-type {
+      type frr-zebra:nexthop-type;
+      mandatory true;
+      description
+	"The nexthop type.";
+    }
+
+    leaf gateway {
+      type frr-zebra:gateway-address;
+      description
+	"The nexthop gateway address.";
+    }
+
+    leaf vrf {
+      type string;
+      description
+	"The nexthop vrf name, if different from the route.";
+    }
+
+    leaf interface {
+      type string;
+      description
+	"The nexthop egress interface.";
+    }
+
+    leaf bh-type {
+      type frr-zebra:blackhole-type;
+      description
+	"A blackhole sub-type, if the nexthop is a blackhole type.";
+    }
+
+    leaf flags {
```

Comment:
Now these are unique values that are only for zebra nexthop information.  Maybe the same approach can be taken here that is being done for route-maps?